### ISSUE TITLE
Possible to skip already optimized SVGs?

### ISSUE THREAD

**Issue by brandondurham** [svg/svgo#767]
I’ve written a script to optimize hundreds of SVGs within a repo when building for production. It runs with every production build and currently re-optimizes every SVG recursively from the root of the repo. This obviously causes a problem considering there are so many to cycle through.

Is there a way for svgo to optimize only the files that have changed since the last build?

**Comment by bil-elmoussaoui** [comment]
I have tried to use svgo recently, and whenever i run the script again on the same file i have a different output. Here's an example
```bash
┏    ~/numix-core  
┗svgo --config=.svgo.yml cantata.svg cantata.svg                 svgo     

Done in 142 ms!
2.778 KiB - 40.6% = 1.649 KiB

┏    ~/numix-core  
┗svgo --config=.svgo.yml cantata.svg cantata.svg                 svgo     

Done in 228 ms!
1.649 KiB - 0.1% = 1.647 KiB

┏    ~/numix-core  
┗svgo --config=.svgo.yml cantata.svg cantata.svg                 svgo     

Done in 163 ms!
1.647 KiB - 0% = 1.647 KiB

```

**Comment by GreLI** [comment]
Yep, in the current implementation several runs can give a bit more optimization. This is why `--multipass` cli-option was added.

But in general how to tell if SVG were optimized? You never know until you'll give a run. There is no sign which can tell for sure. You can't program something without an accurate algorithm.

**Comment by strarsis** [comment]
Ideally `svgo` optimizations are somehow idempotent, which means that after reaching maximum possible optimization (with `svgo`),  the SVG file shouldn't change any further.
Or do you mean a file size threshold or ratio between input and output file size that determines whether to further optimize or not?
